### ISSUE TITLE
fmt should use internal libpng on linux if system libpng is too old

### ISSUE THREAD

**Issue by shakesoda** [HaxeFoundation/hashlink#122]
Currently, if a system uses libpng 1.2 (i.e. it's Ubuntu 16.04 - not sure if they've updated since, I target LTS), fmt throws with "PNG support is missing for this libPNG version"

I ended up just hacking the CMake file to use the internal library (unconditionally) myself, but it should really just detect system version and use its own if that is needed for working support. I'd PR, but my CMake-fu is pretty weak.

Alternate solution would be a libpng 1.2 codepath (which might be reasonable if Ubuntu 18.04 is still using it?)

**Comment by ncannasse** [comment]
I agree that this should be made available, but I'm lacking the time to properly integrate and test the older PNG API which is quite messy compared to png_image higher level API. You're welcome to submit a PR for it ;)

**Comment by ncannasse** [comment]
You can also install libpng16-dev that works well

**Comment by shakesoda** [comment]
I actually can't install libpng16-dev without it conflicting with other libraries I have installed, it flags things like freetype-dev, cairo-dev, gtk3-dev for removal (this was the first thing I tried)

**Comment by andyli** [comment]
Simply build and install your own png version in somewhere (e.g. `/usr/local` or a folder in `~/`), then use `cmake -DPNG_DIR=path ..`, in which `PNG_DIR` pointing to your desired png installation's `PNGConfig.cmake`. See https://cmake.org/cmake/help/latest/command/find_package.html
